File: backend/src/Domain/Entities/Transaction.cs
using Domain.Enums;

namespace Domain.Entities;

public class Transaction
{
    public Guid Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public TransactionType Type { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid PersonId { get; set; }
    public Person Person { get; set; } = null!;
}

File: frontend/src/types/transaction.ts
export const TransactionType = {
  Receita: 0,
  Despesa: 1,
} as const;

export type TransactionType = typeof TransactionType[keyof typeof TransactionType];

export interface Transaction {
  id: string;
  description: string;
  amount: number;
  type: TransactionType;
  personId: string;
  personName: string;
  createdAt: string;
}

export interface CreateTransactionRequest {
  description: string;
  amount: number;
  type: TransactionType;
  personId: string;
}

export interface TotalsResponse {
  totalReceitas: number;
  totalDespesas: number;
  saldo: number;
}
